### PR TITLE
LRQA-57281 | LRQA-57882

### PR DESCRIPTION
**Pull request by gaborlovasdotcom**
Successor of https://github.com/xbrianlee/liferay-portal/pull/460.

https://issues.liferay.com/browse/LRQA-57281
https://issues.liferay.com/browse/LRQA-57882

CC: @dellawang, @xbrianlee

### CODE REVIEW

**Comment by gaborlovasdotcom** [comment]
ci:test:sf

**Comment by gaborlovasdotcom** [comment]
ci:test:relevant

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/oauth2/OAuth2.testcase [hunk 1]
```diff
@@ -671,6 +671,78 @@ definition {
...
+
+		var curl = '''
+			${portalURL}/o/oauth2/token \
+				-d client_id=abc123 \
+				-d client_secret=abc123 \
+				-d grant_type=authorization_code \
+				-d code=${authorizationCode}
+		''';
+
+		echo("curl command ${curl}");
```
Comment (xbrianlee) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/oauth2/OAuth2.testcase:717]: @dellawang Why aren't we using code like this to do this curl post? I'm assuming this is step 7 in https://issues.liferay.com/browse/LRQA-57882?
```
var portalVersionOutput = JSONCurlUtil.post("${portalVersionCurl}");

if (!(contains("${portalVersionOutput}", "7.3"))) {
	fail("JSON-WS call failed to return Portal version");
}
```

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/oauth2/OAuth2.testcase [hunk 2]
```diff
@@ -671,6 +671,78 @@ definition {
...
+			key_activity = "Last Access",
+			locator1 = "OAuth2#USER_CONNECTED_APP_ACTIVITY");
+
+		Pause(locator1 = "17000");
+
+		Refresh();
+
+		AssertElementNotPresent(
+			key_appName = "Test Expired Application",
+			locator1 = "OAuth2#USER_CONNECTED_APP_TABLE_NAME");
```
Comment (xbrianlee) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/oauth2/OAuth2.testcase:743]: @dellawang We need to add a positive assertion before this assertion. The test could throw a NPE and this page could fail to load, but the test would still pass.

**Comment by xbrianlee** [comment]
@dellawang Overall good job on the new tests. Can you incorporate my feedback above and @JoyceWang08211's feedback in the JIRA ticket? I agree with her feedback.
https://issues.liferay.com/browse/LRQA-57281?focusedCommentId=2150737&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2150737

@gaborlovasdotcom Let's make these separate PRs in the future since they're unrelated and they aren't minor changes (new tests are major changes). Also the PR title should follow the format `LRQA-XX, LRQA-YY | master`. The branch name is important so there isn't confusion.

**Pull request closed without merging**